### PR TITLE
add pod support 

### PR DESCRIPTION
**Pull request by aelam**
!! var fullwidthHalfwidthCount: Int is moved out from StudioFoundtion to this one

### CODE REVIEW

**Review by freddi-kit — COMMENTED** [review]
@aelam These sources mainly auto-generates code from python script.  Your modified code is not the same as the one which will be generated by the script. If you want to change code, please change generate script as well.

File: Tests/EastAsianWidthTests/EdgeCharacters+Generated.swift [hunk 1]
```diff
@@ -11,35 +11,35 @@ let EastAsianFullwidthEdgeRawScalarValue: [UInt32] = [
     0x3000,0xff01,0xff60,0xffe0,0xffe6,
 ]
 
-let EastAsianFullwidthEdgeUnicodeScalar: [UnicodeScalar] = EastAsianFullwidthEdgeRawScalarValue.flatMap(UnicodeScalar.init)
+let EastAsianFullwidthEdgeUnicodeScalar: [UnicodeScalar] = EastAsianFullwidthEdgeRawScalarValue.compactMap(UnicodeScalar.init)
```
Comment (freddi-kit) [inline Tests/EastAsianWidthTests/EdgeCharacters+Generated.swift:14]: If you want to modify them, please modify line which `flatMap` in this script code

https://github.com/HevaWu/EastAsianWidth/blob/b70ddd0501b41e6c5dca1cc2c94de5ca315b4698/Scripts/TestGenerator.py#L28

File: Sources/EastAsianWidth/String+Extension.swift [hunk 2]
```diff
@@ -9,45 +9,52 @@ import Foundation
...
+    var containsHalfwidthCharacters: Bool {
         return unicodeScalars.contains { $0.isHalfwidth }
     }
+
+    /// Characters count up by O(n).
+    /// If the character is ASCII code then count is 1, or else count is 2.
+    /// Return value is sum of characters count.
+    var fullwidthHalfwidthCount: Int {
+        return unicodeScalars.countByEastAsianWidth()
+    }
```
Comment (freddi-kit) [inline Sources/EastAsianWidth/String+Extension.swift:59]: Why is it added?

Comment (freddi-kit) [inline Sources/EastAsianWidth/String+Extension.swift:59]: This repo is mainly for only supporting around Unicode things, but this line is specific for Japanese things.  I think we don't need to support it by this project and library use can define it by themselves

**Comment by aelam** [comment]
It was the only dependency in studio foundation so i moved it out
It

freddi(Yuki Aki) ***@***.***> 于 2021年3月25日周四 11:32写道：

File: Tests/EastAsianWidthTests/EdgeCharacters+Generated.swift (hunk 1 above)
Comment (aelam) [reply Tests/EastAsianWidthTests/EdgeCharacters+Generated.swift:14]: oh no.

I don't think it's a good way to generate everyithing with python

File: Sources/EastAsianWidth/String+Extension.swift (hunk 2 above)
Comment (aelam) [reply Sources/EastAsianWidth/String+Extension.swift:59]:
```
public extension UnicodeScalar {
    var isEastAsianFullwidth: Bool {
        switch self.value {
        case 0x3000...0x3000: return true
        case 0xff01...0xff60: return true
        case 0xffe0...0xffe6: return true
        default: return false
        }
    }
```
You see this function. EastAsianWidth is a part of unicode, it's OK to include it

**Review by HevaWu — APPROVED** [review]
LGTM 👍 
Thanks for help updating it! 🙇‍♀️

**Pull request merged**